client/src/main/java/User.java

```diff
@@ -0,0 +1,29 @@
+import java.io.Serializable;
+import java.util.Objects;
+
+public class User implements Serializable {
+    private final String login;
+    private final String pass;
+
+    public User(String login, String pass) {
+        this.login = login;
+        this.pass = pass;
+    }
+
+    public String getNick() {
+        return login;
+    }
+
+    public boolean equals(User o) {
+        if (o == null) {
+            return false;
+        }
+        return this.login.equals(o.login)
+                && this.pass.equals(this.pass);
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [SA: Self comparison of value with itself (SA_LOCAL_SELF_COMPARISON)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-self-comparison-of-value-with-itself-sa-local-self-comparison).
This method compares a local variable with itself, and may indicate a typo or a logic error. Make sure that you are comparing the right things.